docs/mastering-plone/volto_custom_addon2.md

```diff
@@ -293,17 +293,16 @@ You see the Volto `Form` component with its onSubmit event, the form data and th
 :class: toggle
 
 ```{code-block} jsx
-:emphasize-lines: 113-125
+:emphasize-lines: 112-124
```

Comment:
Good catch!

Now if we could only get that code sample to have syntax highlighting. We have noticed that if the syntax is truly not `jsx`, then pygments cannot highlight it properly. Do you see any syntax issue in the example?

Reply:
The lines are hightlighted. Maybe a local problem?

Reply:
The line numbers needed to be shifted by 1. See the preview: https://deploy-preview-626--plone-training.netlify.app/mastering-plone/volto_custom_addon2.html

![Screen Shot 2022-07-10 at 3 45 59 AM](https://user-images.githubusercontent.com/102112/178141645-d1b83290-2306-44e0-bc08-c5aea458ee44.png)

Reply:
Line emphasizing (the yellow one) should work, but JSX syntax highlighting is broken due to the `aria-label` attribute in Line 444:

``` jsx
<button aria-label={props.intl.formatMessage(messages.add)} onClick={addQA}>
   <Icon name={plusSVG} size="18px" />
</button>
``` 

It seems to be the same problem as described here https://github.com/plone/volto/issues/3038#issuecomment-1041390641. Using camelCase (`ariaLabel`) will result in compilation errors. I have no clue how to fix it..  🤔

Reply:
Thank you for identifying the problem. It is in fact a [bug in jsx-lexer](https://github.com/fcurella/jsx-lexer/issues/15#issuecomment-1042808552), which is a plug-in for Pygments which lacks JSX support. Let's leave this code sample as is.

Reply:
I think this may be a simple fix by changing the regex for the [lexer's `Name.Attribute`](https://github.com/fcurella/jsx-lexer/blob/master/jsx/lexer.py#L30).

```python
([\w]+\s*)(=)(\s*)
```

to this:

```python
([\w-]+\s*)(=)(\s*)
```

Reply:
PR submitted: https://github.com/fcurella/jsx-lexer/pull/16

Reply:
I am resolving this bit, because I expect the PR in jsx-lexer to take some time to get merged.

Reply:
Holy cow! Now with a new release!

https://github.com/fcurella/jsx-lexer/releases/tag/v2.0.0